### ISSUE TITLE
Ad list details does not report the invalid domains number

### ISSUE THREAD

**Issue by abesnier** [pi-hole/pi-hole#4441]
### Versions

- Pi-hole:  5.6
- AdminLTE:  5.8
- FTL:  5.11

### Platform

- OS and version:  Debian 10 up to date

- Platform:  x64

### Expected behavior

After running pihole -g or using the admin interface (Tools>Update gravity), the update report shows that some lists contains invalid domains.
Using the Adlist management page (Group Management > Adlists), and by clicking the icon in the first column, the report for the selected adlist shows a line "Number of valid domains on this list", and one "Number of invalid domains on this list".

### Actual behavior / bug

The "Number of invalid domains on this list" does not reflect the status of the gravity update and always shows 0 invalid domains. For example, the list https://raw.githubusercontent.com/dns-test/W3C_annual_most_used_survey_blocklist-master-TOP_EU_US_Ad/master/TOP_EU_US_Ads_Trackers_ABP contains 2111 invalid domains out of 2112, but the Adlist management page shows zero invalid domains.

### Steps to reproduce

1. Run 'pihole -g'
2. Check the update report, and the number of invalid domains per adlist
3. Go to Group Management > Adlists, and check the info for the adlists.
4. Compare the number of invalid domains shown in the UI vs the gravity update report.

### Screenshots

Report from the UI: https://postimg.cc/235Xys6C
Report from pihole -g: https://postimg.cc/ZCGff0Lq

**Comment by abesnier** [comment]
Hi yubiuser, thanks for accepting this report.
I am not good at shell scripts, but I took this issue as an exercise!
From what I saw, this is because of the function `parseList `(gravity.sh > line 509)
`num_invalid `is set to zero just before entering this function (line 508), and then redeclared as local inside `parseList` (line 519), therefore, the global value remains zero when it is used by `database_adlist_number `(line 251).

Cheers
